packages/core/src/Stores/notification-store.js

```diff
@@ -73,8 +75,12 @@ export default class NotificationStore extends BaseStore {
             toggleNotificationsModal: action.bound,
             unmarkNotificationMessage: action.bound,
             updateNotifications: action.bound,
+            p2p_completed_orders: observable,
+            getP2pCompletedOrders: action.bound,
         });
 
+        const debouncedGetP2pCompletedOrders = debounce(this.getP2pCompletedOrders, 1000);
```

Comment:
Debounce the `getP2pCompletedOrders` function to avoid multiple calls in `reaction`.